devito/ops/operator.py

```diff
@@ -30,10 +30,12 @@ def __init__(self, *args, **kwargs):
     def _specialize_iet(self, iet, **kwargs):
         warning("The OPS backend is still work-in-progress")
 
-        # If there is no iteration tree, then there is no loop to be optimized using OPS.
+        # If there is no iteration trees nor affine trees,
+        # then there is no loop to be optimized using OPS.
         iteration_tree = retrieve_iteration_tree(iet, mode='normal')
-        if not len(iteration_tree):
+        if not len(iteration_tree) or not find_affine_trees(iet):
```

Comment:
`find_affine_tree` is already called at line 83

you should hoist it at the top of this method and use the result when needed

Reply:
ok, just re-commited considering this comment. thank you